From c795793c5a914f8f7d31d95df67efeeb38f2a6c1 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 23 Jun 2016 23:33:57 -0400
Subject: [PATCH] IllegalPacketEvent

Fired for invalid data from players that usually represents hacking
---
 src/main/java/net/minecraft/server/PlayerConnection.java | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index e77e6fd9e..c7be44674 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -2231,7 +2231,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                     CraftEventFactory.handleEditBookEvent(player, itemstack1); // CraftBukkit
                 }
             } catch (Exception exception) {
-                PlayerConnection.LOGGER.error("Couldn\'t handle book info", exception);
+                new com.empireminecraft.customevents.IllegalPacketEvent(player.getBukkitEntity(), "InvalidBookEdit", exception).callEvent(); // EMC
                 this.disconnect("Invalid book data!"); // CraftBukkit
             }
         } else {
@@ -2274,7 +2274,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                         CraftEventFactory.handleEditBookEvent(player, itemstack2); // CraftBukkit
                     }
                 } catch (Exception exception1) {
-                    PlayerConnection.LOGGER.error("Couldn\'t sign book", exception1);
+                    new com.empireminecraft.customevents.IllegalPacketEvent(player.getBukkitEntity(), "InvalidBookSign", exception1).callEvent(); // EMC
                     this.disconnect("Invalid book data!"); // CraftBukkit
                 }
             } else if ("MC|TrSel".equals(s)) {
@@ -2286,7 +2286,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                         ((ContainerMerchant) container).d(j);
                     }
                 } catch (Exception exception2) {
-                    PlayerConnection.LOGGER.error("Couldn\'t select trade", exception2);
+                    new com.empireminecraft.customevents.IllegalPacketEvent(player.getBukkitEntity(), "InvalidTrade", exception2).callEvent(); // EMC
                     this.disconnect("Invalid trade data!"); // CraftBukkit
                 }
             } else {
@@ -2428,7 +2428,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                                     iinventory.update();
                                 }
                             } catch (Exception exception5) {
-                                PlayerConnection.LOGGER.error("Couldn\'t set beacon", exception5);
+                                new com.empireminecraft.customevents.IllegalPacketEvent(player.getBukkitEntity(), "InvalidBeacon", exception5).callEvent(); // EMC
                                 this.disconnect("Invalid beacon data!"); // CraftBukkit
                             }
                         }
@@ -2528,7 +2528,8 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
                             this.player.playerConnection.sendPacket(new PacketPlayOutSetSlot(-2, k, this.player.inventory.getItem(k)));
                             this.player.playerConnection.sendPacket(new PacketPlayOutHeldItemSlot(this.player.inventory.itemInHandIndex));
                         } catch (Exception exception7) {
-                            PlayerConnection.LOGGER.error("Couldn\'t pick item", exception7);
+                            new com.empireminecraft.customevents.IllegalPacketEvent(player.getBukkitEntity(), "InvalidPickItem", exception7).callEvent(); // EMC
+                            this.disconnect("Invalid PickItem"); // EMC
                         }
                     }
                     // CraftBukkit start
-- 
2.11.0

